### DISCUSSION TITLE
Increase the maximum log size - currently it is 4KB

### DISCUSSION THREAD

**Discussion by JCMais** [vercel/community#1012] [Idea]
From https://vercel.com/docs/concepts/limits/overview

![chrome_2022-10-31_16-21-15](https://user-images.githubusercontent.com/1041647/199092797-4506ac4e-7e16-4238-a575-2a36222ea676.png)

While debugging something, it is pretty easy to go over the limit, which causes a lot of useful information to get lost, as it only retains the last 4kb.

It seems that the limit also applies to log drains, as we are using the Datadog integration and the logs in there are also missing a lot of lines.

**Comment by alfonso-paella** [comment]
It's hard to run a real production app with this limit in place 😅

**Comment by statico** [comment]
I understand that I should be using the +1 button, but this is a frustrating limitation that has really hindered the debugability of our app in production.

**Reply by statico** [reply]
We're already paying for Logtail (which is great) which gets logs via the Vercel log drain, so I'm going to rewrite our logger to use Pino to log to Logtail directly: https://betterstack.com/docs/logs/javascript/

AMA

**Reply by statico** [reply]
This has been running for a few months. Unfortunately, Logtail's log flushing sporadically fails, which results in some amount of logs being logged.

We really need a good solution for logging from Vercel. It's crucial in order to debug production issues.

**Reply by alexkvazos** [reply]
On this same train.

**Comment by alfonso-paella** [comment]
We did that on datadog, but it's not great. I believe vercel injects some
headers in their proxy that you don't have access to from application code.

I may be wrong / would love to be corrected

**Comment by jhb-dev** [comment]
It seems vercel increased the log limit: https://vercel.com/blog/evolving-vercel-functions#increased-logging-limits

**Reply by bholagabbar** [reply]
> It seems vercel increased the log limit: https://vercel.com/blog/evolving-vercel-functions#increased-logging-limits

Were you able to figure out how to increase the limit for your app ? cc @timpetri

**Comment by timpetri** [comment]
I'm still observing this limit applied both in my own Vercel Logs and Log drain target. The documentation around this limit hasn't been updated since before the announcement post. Does anyone know where this can be tweaked?

**Comment by bholagabbar** [comment]
Any update here ?

**Comment by n-vigier** [comment]
Hi all,

I have confirmed with the Docs team that the docs already reflect the change. Previously and with non-streaming functions the logs are limited to 4KB in total.
Now it’s 4KB per line, which is what the blog post also mentions.
A workaround to increase the limit is to split the content you want to log out into smaller chunks.

For example, instead of console logging an entire array, you can loop over the array and log the individual items. This will produce separate Runtime Logs entries for each invocation of `console.log`, each with a 4KB limit.
